pkg/liquidity-mining/contracts/veBALDeploymentCoordinator.sol

```diff
@@ -0,0 +1,274 @@
+        //
+        // The BalancerTokenAdmin contracts needs admin permission over BAL in order to complete this process, and we
+        // need to be authorized to make the call.
+        _balancerToken.grantRole(_balancerToken.DEFAULT_ADMIN_ROLE(), address(_balancerTokenAdmin));
+        authorizer.grantRole(_balancerTokenAdmin.getActionId(IBalancerTokenAdmin.activate.selector), address(this));
+        _balancerTokenAdmin.activate();
+        // Balancer Token Admin activation automatically removes authority over the BAL token from all other accounts,
+        // so we don't need to renounce this permission.
+
+        // Step 2: grant BalancerMinter authority to mint BAL, as part of the Liquidity Mining program.
+        authorizer.grantRole(
+            _balancerTokenAdmin.getActionId(IBalancerTokenAdmin.mint.selector),
+            address(_balancerMinter)
+        );
+
+        // Step 3: setup the Liquidity Mining program by creating the different gauge types on the Gauge Controller.
+        //
+        // All gauge types are initially created with a weight of 0, which will allow for gauges to be deployed and LPs
+        // to vote and stake for them, without yet minting any BAL. This is intended to provide a grace period for LPs
+        // to migrate to the new system before it is fully activated. The gauge type weights will be set to their actual
+        // values on the last stage of deployment.
+        {
+            // Admin functions on the Gauge Controller have to be called via the the AuthorizerAdaptor, which acts as
+            // its admin.
+
+            // Note that the current Authorizer ignores the 'where' parameter, so we don't need to (cannot) indicate
+            // that this permission should only be granted on the gauge controller itself.
+            authorizer.grantRole(authorizerAdaptor.getActionId(IGaugeController.add_type.selector), address(this));
+
+            _addGaugeType(_gaugeController, "Liquidity Mining Committee");
+            _addGaugeType(_gaugeController, "veBAL");
+            _addGaugeType(_gaugeController, "Ethereum");
+            _addGaugeType(_gaugeController, "Polygon");
+            _addGaugeType(_gaugeController, "Arbitrum");
+
+            authorizer.revokeRole(authorizerAdaptor.getActionId(IGaugeController.add_type.selector), address(this));
+        }
+
+        // Step 4: create gauges for the single-gauge gauge types
+        //
+        // The LM committee and veBAL gauge types will have a single gauge of the single-recipient gauge kind
+
+        // grant self power to add gauges
+        // deploy lm, vebal, merkle gauges. add gauges to gauge controller
+
+        // grant gauge adder authority over controller
+        // grant self power over guage adder
+        // add l1 and l2 factories to gauge adder
+        // deploy l1 and l2 gauges, add to controller via gauge adder
+
+        // add tokens to l1 and l2 gauges
```

Comment:
I have a local branch which implements the setup of a [GaugeAdder](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1158) contract, allowlist the factory contract for Ethereum gauges and deploy gauges for a preset array of pools.

The comments here mention adding factories for L2 gauges and deploying single recipient gauges. L2 gauges are definitely out of scope for being deployed in stage 1 but we can potentially add in single recipient gauges as well.

Reply:
I'm not sure what gas costs might end up being if we're deploying a large number of gauges here (granted these are proxies). It might be worth splitting off step 4 to be another function call which is immediately callable by anyone after `performFirstStage` has been called.